src/pkg/engine/pepr/stream_test.go

```diff
@@ -0,0 +1,394 @@
+package pepr
+
+import (
+	"bytes"
+	"io"
+	"regexp"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestNewStreamReader(t *testing.T) {
+	timestamp := true
+	filterNamespace := "test-namespace"
+	filterName := "test-name"
+
+	reader := NewStreamReader(timestamp, filterNamespace, filterName)
+
+	assert.True(t, reader.showTimestamp)
```

Comment:
Generally speaking, we prefer `require` over `assert` to ensure the test fails immediately, but if you think `assert` is appropriate here I'm to that as well